File: _legacy/controller/users/delete_item.php
<?php
namespace Core;
use Exception, Controller;

if (!defined('__API_GOOSE__')) exit();

/**
 * delete user
 *
 * @var Goose|Connect $this
 */

try
{
  // check and set srl
  $srl = (int)($this->params['srl'] ?? 0);
  if ($srl <= 0)
  {
    throw new Exception(Message::make('error.notFound', 'srl'), 404);
  }

  // connect db
  $this->model->connect();

  // check data
  $cnt = $this->model->getCount((object)[
    'table' => 'users',
    'where' => 'srl='.$srl,
  ])->data;
  if ($cnt <= 0)
  {
    throw new Exception(Message::make('error.noData', 'user'), 204);
  }

  // check authorization
  $token = Auth::checkAuthorization($this->model, 'admin');
  if (!$token->data->admin && ((int)$token->data->srl !== $srl))
  {
    throw new Exception(Message::make('error.access'), 401);
  }

  // remove item
  $output = Controller\Main::delete($this, (object)[
    'table' => 'users',
    'srl' => $srl,
  ]);

  // TODO: 사용자를 지우면 컨텐츠들을 지워야 할거 같은데 이건 좀 고려해보자..

  // set output
  if ($token) $output->_token = $token->jwt;

  // disconnect db
  $this->model->disconnect();

  // output data
  return Output::result($output);
}
catch (Exception $e)
{
  if ($this->model ?? false) $this->model->disconnect();
  return Error::result($e->getMessage(), $e->getCode());
}

File: src/endpoints/nest/delete_item.py
from . import __types__ as types
from src import output
from src.libs.db import DB, Table
from src.modules.verify import checking_token
from ..category import __libs__ as category_libs

async def delete_item(params: dict = {}, req = None, _db: DB = None, _check_token = True):

    # set values
    result = None
    db = _db if _db else DB().connect()

    try:
        # set params
        params = types.DeleteItem(**params)

        # checking token
        if _check_token: checking_token(req, db)

        # check item
        count = db.get_count(
            table_name = Table.NEST.value,
            where = [ f'srl = {params.srl}' ],
        )
        if count == 0: raise Exception('Item not found.', 204)

        # delete category
        category_libs.delete(db, category_libs.Module.NEST, params.srl)

        # TODO: 추가로 작업 해야할것들
        # TODO: - 아티클 데이터 삭제
        # TODO: - 아티클에 해당되는 파일 삭제
        # TODO: - 아티클에 해당되는 댓글 삭제

        # delete item
        db.delete_item(
            table_name = Table.NEST.value,
            where = [ f'srl={params.srl}' ],
        )

        # set result
        result = output.success({
            'message': 'Complete delete nest.',
        })
    except Exception as e:
        result = output.exc(e)
    finally:
        if not _db and db: db.disconnect()
        return result
